CMake Build System

WhoBrokeTheBuild opened the pull request.

WhoBrokeTheBuild committed:
WIP CMake

Include previously bootstrap-generated mdsconfig.h.in

Add running of deploy/yylex.gen to CMake

Wrap JavaMds in IF(ENABLE_JAVA)

Fix link in MdsObjectsCppShr s/JavaMds/MdsIpShr/g

WIP CMake
Lowercase all of the CMake functions
Handle BUILD_SHARED_LIBS=OFF
Add HDF5

WIP CMake
Refactoring and standardization
Moved LIBXYZ variables into FindXYZ packages

WhoBrokeTheBuild committed:
Refactor/Cleanup, Replace yylex.gen
Standardize all the FindXYZ scripts
Fix the manifest warnings from fat_jar()
Replace deploy/yylex.gen with explicit foreach loops
Replace tdishr/yylex/TdiHash.gen with TdiHash.py
Move *.y and *.l files out of yylex subdirectories

WhoBrokeTheBuild committed:
WIP CMake
Remove conf/
Start work on CMakeMigrationGuide
Convert opcbuiltins.h to opcodes.csv
Generate opcbuiltins.h with opcbuiltins.py
Generate TdiHash.c from opcodes.csv
Comments, cleanup, refactoring

WhoBrokeTheBuild committed:
Fix issues found while comparing new installation with old

Make liblzma and git submodule update optional

Refactor how files are generated, prefix gen-
Add opcode number to opcodes.csv
(No longer rely on the line number)
Temporarily remove git submodule update

Rename gen_ file. Refactor gen_compound. Add CMakeLists

Fix broken opcodes.csv

WIP

Add missing SetEnvironmentForTests
Add clangd's .cache/ to gitignore
Comment out unused tests for LabView
Add __attribute__((unused)) to help compile with newer compilers

dgarnier committed:
Feature/cmake macos (#2)

* Finish reinstating bootstrap

* fix type compatibility when compiling with Intel Fortran (ifort)

* enable building with Intel Fortran

* Add rpath functionality to remove dependence on LD_LIBRARY_PATH

fix install prefix in rpath

* Fix broken CMake 3.24 FindJNI on MacOS.  Also allow JDK version setting on MacOS.

* Fix use defined CACHE variable LIBXML2_DIR

* fix allowing ENABLE_IDL=OFF to work when IDL is not on host

* silence unused but set warning (+1 squashed commit)
Squashed commits:
[1c4dde515] TdiMath1.c scalars set but not used warning.  Noop use inserted (because we don't yet have C23 and [[mabye-unused]] attribute.)

* Darwin namespace collision avoided with ensureing _POSIX_C_SOURCE

* testing fixes for MacOS (+1 squashed commit)
Squashed commits:
[6798c3ff4] fix name collision in UdpEventsTest on macos

* change python setup to work with newer setuptools

* improve FreeTDS/Sybase support

* fix labview headers for newer macos compatibility

* change sprintf to snprintf for gcc12 warnings

* testing udp events in parallel leads to false error

* testing on MacOS does not require unavailable librt

* don't rely on dynamic linking in mitdevicesio

* Missing links I think should be correct.

* MacOS, prefer path python over system python

* Allow fixed link of MacOS tdic instead of broken module TdiShr

---------

Co-authored-by: Stephen Lane-Walsh <slwalsh@psfc.mit.edu>

dgarnier commented:
The current ./bootstrap is not actually working.  It makes .c files from .l and .y in the source tree, but the cmake files expect them to be in the build directories.  I have a branch that replaces the bison/flex part of bootstrap.  I did have to make small changes to the .l and .y files though, because of the hardwiring of the output files inside those input files.  See: https://github.com/dgarnier/mdsplus/tree/feature/cmake_bison_flex

WhoBrokeTheBuild committed:
Completely rework bootstrap and gen-* scripts
Rewrite bootstrap in python
Move gen-* scripts back to deploy
Remove argparse in favor of constants
Add more verbose output
Replace os.walk with glob.glob in gen-messages
Rename gen-* scripts to be more explicit
Remove old CMake code that was intended to replace bootstrap
Add CMake code to run bootstrap if it hasn't already been run

WhoBrokeTheBuild committed:
Add __attribute__((unused)) to squash clang warnings

Set C/C++ Standard, Add ENABLE_LABVIEW

Re-enable auto git submodule update, which may cause problems

Add a first pass at running Java Tests

Add mdsplus-api Java tests as well

Add tests for tditest, small fixes / cleanup

Now with C++ tests!

Fixes for tests
Fix py-task-do because it actually found an error!

Fixes for python and java tests
We now respect $PYTHON and try to use that version for scripts and $PyLib

Turn $PYTHON into -DPYTHON, and have it default to $PYTHON

Rehaul testing, work on Java
Add mdsplus_add_test to wrap all calls to add_test
This sets the environment variables and allows for defaults
Java testing works, but isn't great

Check for `FAILED` in the test output to catch the python failures
Add main/subtree paths to test environment specifications

WhoBrokeTheBuild committed:
Add support for valgrind
When ENABLE_VALGRIND=ON, all tests will be generated both normally, and with every available valgrind tool
Because mdsplus_add_test() now calls add_test() multiple times, it can now output the list of targets into ${TEST_LIST_VARIABLE}
This can be overridden for a test by specifying NO_VALGRIND
TODO: allow disabling specific tools
Valgrind suppression files and flags are passed through properly
TODO: sanitizer support in a similar fashion
Add the testing/selftest directory, which was *not* included in the autotools build system, but seems to have some useful stuff
We do not currently support TAP exit codes like 77 or 99, this could be added in the future

WhoBrokeTheBuild committed:
Add .vscode/launch.json generation, WIP valgrind tests
You can now pass `-DGENERATE_VSCODE_LAUNCH_JSON=ON` and every test will become an entry in `.vscode/launch.json`
Added a fix for clang >= 14 and valgrind < 3.20
Added TEST_INDEX for multiple runs of the same test, started updating tests to use it to offset port numbers and shot numbers
Fix HAVE_VALGRIND_H in mdsconfig.h.in

Properly configure LIBPATH in envsyms

WIP RPATH Changes

WhoBrokeTheBuild committed:
Remove bootstrap, 3rd-party-apis, Refactoring
3rd-party-apis now gets downloaded by CMake into the build folder
Prefix all CMake includes and macros with `MDSplus`/`mdsplus_`
Merge GetDate into MDSplusGitRevision
Replace bootstrap.py with MDSplusGenerateFiles
Make sure each gen-* script chdir's to the repository root
Remove explicit GCC check for ENABLE_GPROF, now check for -pg flag directly

WhoBrokeTheBuild committed:
WIP Valgrind support
Upped default TEST_TIMEOUT to 1500 (the same as ctest)
Work on adding $TEST_INDEX to each test in order to offset shot/port numbers

WhoBrokeTheBuild committed:
CMake Refactor
Rework compiler/linker flag handling into macros: mdsplus_check_compiler_flag, mdsplus_check_linker_flag
Set Java -source version
Add include_guards for cmake files
Squash warnings from clang 18

WhoBrokeTheBuild committed:
Cleanup and refactor
Replace ENABLE_TESTS with the built in BUILD_TESTING
Cleanup .gitignore, remove all intermediate files and outdated files/directories
Rename all files that would otherwise be renamed during install
Copy scripts to build/bin/ in addition to installing them, to make the build directory more usable

WhoBrokeTheBuild committed:
Add `./configure` script to help with migrating to cmake

Fix broken BUILD_SHARED_LIBS=OFF

WhoBrokeTheBuild committed:
Major overhaul of cmake options and flag checking
Add mdsplus_option() macro to unify option()/set(CACHE STRING) options and messaging
Rework mdsplus_check_flags to allow testing multiple compiler/linker flags at once (needed by the sanitizer)
Rename ENABLE_PERF to ENABLE_TREE_PERF
Remove USE_PERF from mdsconfig.h.in
Add USE_TREE_PERF define in CMake instead
Remove JAVA_TARGET_VERSION and JAVA_BOOTCLASSPATH, they were removed in Java 9
Remove Java -source for the same reason
Add JAVA_VERSION to set a minimum version for find_package(Java) and to pass to java_home on OSX
Standardize calls to cmake_parse_arguments
Convert cmake macros to functions, where possible
Standardize documentation for cmake functions

WhoBrokeTheBuild committed:
Begin work on the CMake Developer GUide

Small tweaks and fixes
Fix FindIDL on OSX
Fix FindJava on OSX

Use MDSPLUS_LIBRARY_PATH for do_tditests
OSX can filter out [DY]LD_LIBRARY_PATH, which causes issues

Fixes, tweaks, and refactoring

Remove packaged zlib with mdstcpip, we require the system zlib anyway
Refactor dependency checking to use `if(TARGET XYZ)` instead of `if(XYZ_FOUND`
Add the following options:
ENABLE_MOTIF
ENABLE_SYBASE
ENABLE_HDF5
These were already optional, but now you can stop CMake for even looking for them
Fix issues if Motif is not found/requested
Remove a few `file(GLOB`
Add the ALIAS back to `mdsplus_add_static_copy` since a few libraries link against the static versions directly

WhoBrokeTheBuild committed:
Add initial version of build.py
Disable in-source builds for CMake
If ENABLE_[MOTIF, SYBASE, HDF5] are set, then REQUIRE those packages

WIP Java, build.py
Remove fat_jar()
Add testing and valgrind support to build.py
Add hack for jScope and comments to maybe improve it in the future
Add opts files with new docker images

WIP Tests
Move test variations (memcheck, helgrind) into their own working directories
Set default_tree_path on all tests to be .
Add MDSIP_CLIENT_LOCAL_LOGFILE workaround
Remove TEST_INDEX modifying shot numbers, it had caused an issue with -1 and model options
build.py now properly kills test subprocesses when Ctrl-C'd

WIP Testing
Change TEST_INDEX from an offset from the base case (1 for memcheck, 2 for helgrind, etc) to a global index, so that every test can make unique ports
Update Java valgrind suppressions
Move Java test arguments into a variable and add comments
More changes to help the tests run in parallel

Fix: UdpEvents tests now run in parallel on macOS

Small fixes
build.py is now actually python 3.6+ compatible
CMAKE_INSTALL_PREFIX is better handled

WIP build.py now supports sanitize
The packaged blas will now actually build, although maybe we can remove this
The Python tests are now skipped if a sanitizer is enabled, we should fix this too
Removed the GUIs as required variables in FindMDSplus

Fix generator logic, add amazonlinux

Jenkins support, Rework deploy/opts/*
Small fixes for build.py

Fix build.py to check the result codes of subprocesses

The script will now stop when docker / CMake fail, and properly report that error code to the user.

WIP Test isolation

Remove mulit-build from build.py
Add MDSPLUS_TEST_INDEX_OFFSET to allow a default value for TEST_INDEX
Set MDSPLUS_TEST_INDEX_OFFSET in the Jenkinsfile to 1000*$EXECUTOR_NUMBER
Move archiveArtifacts to post { always { } }

Add missing -D

Remove --test from test-*.opts

Replace EXECUTOR_NUMBER with OSList.indexOf

Fix argument passing to docker

WIP JUnit support

WIP JUnit support

WIP JUnit support

WIP JUnit support

Update Python tests using TEST_INDEX

WIP LD_PRELOAD/Sanitizers

Add support for docker networks to build.py

Isolate tests with docker networks, instead of port ranges
TODO: mdsevent_port

Rework all port numbers in tests

Ports are now registered in testing/ports.csv, and allocated to tests as needed
This keeps tests from using the same ports when run in parallel
MDSplusAddTest now properly sets a unique mdsevent_port for each test
This keeps the UDP event messages from colliding with each other
MDSplusAddTest also sets TEST_PORT_OFFSET to multiples of 1000 for each test variant (memcheck, helgrind)
This meant that TEST_INDEX could be removed from almost every test, and replaced with TEST_PORT_OFFSET where applicable

Fix Jenkins truncating JUnit logs

Update CMake to output the computed configuration variables, not just the user supplied
This means it will now show the PYTHON, JAVA_HOME, and JAVA_VERSION that it found

Increase sleep time in C++ MdsEventTest

Improve build.py

Add support for aliases for --os, and list the available options in --help
Change the default directory structure generated by build.py
* manual builds should use `build/`, this is no longer used by build.py
* local builds now use `workspace/`
* --os builds now use `workspace-{--os}/`

Within those workspace folders, by default there is:
* `build/` for the intermediate files
* `install/` for the installed `usr/local/mdsplus` files

If you run interactively, there are also helper scripts
* `do-configure.sh` to run the configure step
* `do-build.sh` to run the build
* `do-install.sh` to run the install
* `setup.sh` to set MDSPLUS_DIR and source $MDSPLUS_DIR/setup.sh

--install is now always enabled, and simply defaults to workspace/install

Attempt to filter out the MDSplus-stuff from the environment when running an interactive shell when $MDSPLUS_DIR is detected

Support for Clang 18 and GCC 13

Deprecate TreeNOTOPEN in favor of TreeNOT_OPEN
Increase test timeout to 1h
Add opts for Ubuntu 24

WhoBrokeTheBuild committed:
Update Jenkinsfile, fix embedded version tag

WIP Fixing Jenkins/CMake

WIP Fixing Jenkins/CMake

WIP Fixing Jenkins/CMake

Fix tditest/test-tab for alpine

WhoBrokeTheBuild committed:
WIP CMake

Add --test-regex/-R to build.py to run specific tests matching a regex
Add support for mdslib_ftest and mdslib_fremotetest
Add the somewhat broken task_case back to the python tests
Remove ubuntu24.opts, instead use ubuntu-24

Convert tdishr.h into a generated file

To better support IDE's and syntax highlighting, it's better to just generate all the lines of tdishr.h directly, instead of using the `#include <opcbuiltins.h>` trick

WIP Packaging

Add hard-coded support for debian packaging in build.py
Small refactors in /CMakeLists.txt

Fix fortran flags for mdslib and python detection for cmake

WIP Windows builds for CMake

Disable portions of the build and tests that do not run on windows
Replace all #cmakedefine01 with #cmakedefine in mdsconfig.h.in, and standardize all references to the defines they set
Add an INTERFACE IMPORTED library for labview on windows
Add TEST_WITH_WINE and WINE_PYTHON options for running tests with wine
Add a hack in build.py to use mingw64-cmake for windows builds
Remove windows.env and add new options to windows.opts
Create a wineprefix in ${CMAKE_BINARY_DIR}/.wine when TEST_WITH_WINE is specified
Use $<TARGET_FILE:${name}> for all tests so that they work when called with wine
Add NO_WINE option to mdsplus_add_test to prevent calling wine for a given test

Add notes and TODOs

Refactor build.py, WIP windows build and packaging

Rename .opts files and symlinks to include the architecture as defined by the target system
e.g. amd64 for debian and x86_64 for redhat
Update debian, redhat, and windows installation scripts
Add PLATFORM to control how files are installed in preparation for being packaged
Split windows installer into two separate installers for win32 and win64, this will need some additional work
Fix missing JNI on windows by patching in the 3rd-party-apis
Add MDSplusGetAllTargets cmake function to support windows library conversion using gendef/dlltool
Force Java to compile with compatability for Java 8 (1.8)
Actually turn on HDF5, which was not even included
Add jars for DeviceBeans and jDevices

Fix .vscode/launch.json generation

This is now done at build-time with `make generate-vscode-launch-json` which can properly resolve the `$<TARGET_FILE:_target>` statements from CMake
The --setup-vscode option in build.py has been updated to handle this

Update opts naming

Update opts files to reflect system arch naming conventions

Work on windows packaging

Finish splitting the x86/x64 windows kits
Remove the unnecessary bin_x86/bin_x86_64 config for the windows platform

WhoBrokeTheBuild committed:
Cleanup after rebase

WhoBrokeTheBuild committed:
Fix Jenkinsfile

Jenkinsfile fixes and disable java tests for wine

Reorder build.py so that --test can report an error status
Remove amazonlinux from the Jenkinsfile

Disable symlinks for archiveArtifacts

Otherwise this hangs forever when following symlinks in .wine

Quick attempt at adding --os=bootstrap

WhoBrokeTheBuild committed:
Add try/catch for junit parsing in the Jenkinsfile

heidthecamp committed:
Readd support for rhel 7 and debian 10

Remove 32 bit packages for rhel7

WIP Jenkins

Add Ubuntu 24.04, RHEL 7 and Debian 10 to the Jenkins OS list
Prevent mdslib_ftest and fremotetest from modifying default_tree_path or using the same tree
Disable py-task-do until we can fix it

WIP Jenkins

Revert --bootstrap in build.py
Revert Jenkinsfile to individually clone/bootstrap for each OS (now with retries)
Patch testing for tditest

WIP Jenkins

WIP Jenkins

WIP Jenkins

WhoBrokeTheBuild committed:
Fix readline.dll segfault for windows

Enable test-tab for tditest

WIP Jenkins

Work to better support sanitizers
Disable Java tests when a sanitizer are active

santorofer committed:
Add Arm64 for ubuntu24 to the build system

Add ARM64 for Ubuntu

Add ubuntu-24-arm64 to the OSLisst in the Jenkinsfile

WhoBrokeTheBuild committed:
Cleanup, Add vscode launch targets for common tools

Fix issues with the interactive environment from build.py
Update tdishr.h generation to include the definitions from the Apple Silicon port
Update the address sanitizer to use Ubuntu 24.04
Add vscode launch targets for some common tools:
* mdstcl
* python
* tdic / tditest
* mdsip (on port 8888)

heidthecamp committed:
Enable MATLAB testing in Jenkins

Set default for ENABLE_SYBASE to ON for all platforms except windows

heidthecamp committed:
Harden version parsing in build.py

heidthecamp committed:
Fix windows x84 build

Up dated build to correctly call libraries for x86 or x64 when packaging

WhoBrokeTheBuild committed:
WIP Publishing

Add package verification for debian and redhat
Fix installation bugs that caused package verification to fail
Remove tests from installed python kits
Add `publish.py` which calls to `deploy/platform/PLATFORM/PLATFORM_publish.py` with a given `dist` directory and `mdsplus-publish.json`
Add `debian_publish.py`, `redhat_publish.py`, and `windows_publish.py`, and a stub for `macosx_publish.py`
Move the package signing logic into these new publish scripts, meaning packages will now be built but not signed until they are published
Remove the .build-id artifacts from redhat builds
Fix inconsistency in windows packaging when using $PLATFORM and $DISTNAME
Add jDispatcher to the build

Fix diff garbage in create_installer.sh

Remove python tests from windows installer

Fix windows installer again

chmod +x deploy/publish.py

Update paths for mdsplus-publish.json in the Jenkinsfile

Hopefully run publish.py in a tty

Remove -it from the docker_args in publish.py

Update publishing scripts

Replace shutil.copytree with subprocess calls to rsync for portability
The lowest version we support should be python 3.6, so f-strings should be fine everywhere

Add a hack to avoid using `other` as the flavor while testing

Move the cmake -> alpha flavor hack to build.py

Add debug prints

Hopefully fix debian publish

Add code for not publishing over current releases

Add cleanWs to the head node's setup

Put the new release version files in the right place

Remove wildcard from subprocess call

santorofer committed:
Add Debian 12 arm64

santorofer committed:
s/amd64/arm64/g debian-12-arm64.opts

WhoBrokeTheBuild committed:
Changes from Final Review Process

Added comments to explain confusing things found during the review process
Remove outdated TODOs
Fixed typos
Add automatic PLATFORM detection
Rewrite all incorrect `*_ROOT_DIR` variables to `*_ROOT`
Minor but frequent refactors
Remove -D from all calls to `target_compile_definitions()`
Actually check if `__attribute__((__visibility__))` is supported to define `HAVE_VISIBILITY`
Add a newline at the end of every cmake file
Align the `CMAKE_*` variable displays
Move Sybase library hack into mdssql/
Shuffle around Java cmake variables
Remove `mdsplus_add_compiler_flag()`
Rewrite`mdsplus_check_add_flags()`
Remove `HAVE_VALGRIND_H` and all things associated
Remove references to `HAVE_CONFIG_H`

WhoBrokeTheBuild committed:
Disable flaky wfevent tests

WhoBrokeTheBuild committed:
WIP Build System Review

Reorder build.py
Cleanup arguments for build.py
Rewrite awkward double-argument parsing for .opts files in build.py
Add $DISPLAY to the passthrough environment variables for interactive mode
Protect against missing env vars when spawning interactive mode
Add lots of comments and clarifications
Rework RELEASE_DATE into BUILD_TIMESTAMP
Rewrite MDSplusGitRevision into MDSplusProjectVersion and make it more resilient against not having git

Missed part of build.py

WhoBrokeTheBuild committed:
WIP Updates from review process

Add --flavor and --version arguments to build.py as a front for -DRELEASE_TAG in CMake
Remove git from build.py entirely, replace with RELEASE_BRANCH and RELEASE_VERSION values from CMake cache
Add MDSPLUS_COFNIGURE_COMPLETE CMake cache variable and use it to automatically pass --configure if the previous configure failed
Remove unnecessary --toolchain from build.py
Cleanup and improved error handling

Cleanup more TODOs

Add tests for isatty() when using --tty with docker and console colors
Add more error handling
Add explicit check for python3 in a docker image before using it
Add check to fake setup.sh to prevent use when /etc/mdsplus.conf or $HOME/.mdsplus are present, this is because we can't guarantee an isolated environment if those config files modify it
Refactor do_build to always call cmake --build, which allows us to use --clean-first when the user specifies --clean
Refactor passed_tests and failed_tests into finished_tests

Cleanup TODO comments

WhoBrokeTheBuild committed:
CMake support for macosx

Add `deploy/os/{--os}.env` / `--env-file` support to build.py
Add find script for LibFFI
Add USE_LIBFFI which defaults to ON for ARM64 APPLE
Add deploy/os configs for homebrew and macports
Add conditional MACOS_ARM64 to mdsconfig.h.in
Small tweaks and fixes found during development

WhoBrokeTheBuild committed:
Changes from final review

Small tweaks and typo fixes
Add more error handling
Replace `which` with `command -v` in most places for broader compatibility

WhoBrokeTheBuild committed:
WIP Jenkins

Throttle builds with -j8
Be more resilient if the docker network already exists

Actually change -j this time

WhoBrokeTheBuild committed:
Add --tty back to do_docker in build.py

WhoBrokeTheBuild committed:
Better error handling around 3rd-party-apis download

WhoBrokeTheBuild committed:
Temporarily add cmake to debian repo/conf/distributions

WhoBrokeTheBuild committed:
Fix issues found by --verify-packages

Add -DGSI=gcc64 to all amd64 platforms
Only include `macosx/` on apple
Remove spaces in `Put Data.vi` filenames
Replace `cmake` branch name with `alpha` to test packaging scripts

WhoBrokeTheBuild committed:
Readd MDSplus-previous to debian publish

Readd call to `reprepro clearvanished` in `debian_publish.py` to prevent the release from failing if the distributions file doesn't match the contents of the repo

Fix typo

WhoBrokeTheBuild committed:
Fully enable Jenkins OSs

Enable MacOSX homebrew/macports
Enable Address, Thread, and Undefined Behaviour sanitizers
Enable Valgrind's Memcheck and Helgrind

WhoBrokeTheBuild committed:
WIP macosx support

Fixed issues with --env-file in build.py

Further harden the XML generation

Replace not just BEL, but all non-printable characters from the stdout of tests while writing the jUnit XML

WhoBrokeTheBuild committed:
Enable mit devices for osx, needed by test-dev-py

WhoBrokeTheBuild committed:
Make raw1394 an optional dependency

WhoBrokeTheBuild committed:
WIP Test port isolation for macosx

WhoBrokeTheBuild committed:
Don't try to package test-* OSs

WhoBrokeTheBuild committed:
WIP isolating ports for macosx

WhoBrokeTheBuild committed:
Remove ENV from `$ENV{TEST_PORT_OFFSET}`

WhoBrokeTheBuild committed:
Harden download of 3rd-party-apis

WhoBrokeTheBuild committed:
Move /tmp/test-mdsip.hosts to ./test-mdsip.hosts

WhoBrokeTheBuild committed:
Fix test-mdsip.tdi

WhoBrokeTheBuild committed:
Fix test-mdsip, and missing $PATH issue in interactive mode

WhoBrokeTheBuild committed:
Fix tsan errors, kind of

Move the sanitizers back to redhat, as `/bin/sh` on Debian/Ubuntu segfaults when used with `LD_PRELOAD=tsan.so`
Add support for comments in the .opts files to document this ^
Improve the test summary in build.py when using a test regex

WhoBrokeTheBuild committed:
Skip publishing for test-* OSs

WhoBrokeTheBuild committed:
WIP OSX Packaging & Publishing

Rewrite {platform}_publish.py scripts to accept --release-dir, --publish-dir and --cert-dir instead of hardcoding /release, /publish, and /sign_keys
Update macosx-homebrew to use just "homebrew" as the --distname
Update macosx-macports to use just "macports" as the --distname
Update build.py to copy the tar of `install/usr/local/mdsplus` into the dist dir for macosx

WhoBrokeTheBuild committed:
Fix glob in macosx publish

WhoBrokeTheBuild committed:
WIP OSX Packaging

mwinkel-dev commented:
I have perused the 343 files changed by this work-in-progress PR and didn't see anything alarming.   This high-level review confirmed that the changes are in the expected categories (Jenkins build scripts, CMake scripts, tests, environment variables, platforms, generation scripts, removing obsolete files, etcetera).   With such a large PR, there could of course be some overlooked bugs.   However, the risk of that is low because testing apparently shows that this CMake PR does build on many platforms and the resulting MDSplus does pass the automated test suite.   Thus, if there are overlooked bugs, they are likely minor and thus should be easy to fix after this PR is eventually merged.

WhoBrokeTheBuild committed:
WIP Jenkinsfile

Move over AdminList from alpha Jenkinsfile
Move over options and triggers from alpha Jenkinsfile
Move over commented-out sections that we will need from the alpha Jenkinsfile

WhoBrokeTheBuild committed:
WIP Jenkinsfile

mwinkel-dev reviewed (dismissed):
I have also reviewed the two commits made after my previous post.   Those commits also look reasonable, so am approving this PR.

WhoBrokeTheBuild committed:
Cleanup from PR review

Add fake bootstrap as to not break existing workflows

mwinkel-dev reviewed (dismissed):
The last commit has three minor changes:

1. Bootstrap script for backwards compatibility for people used to the old build scripts.
2. Deletion of extraneous printf() statements.
3. Removal of an obsolete file.

All looks good, so approving this PR.

WhoBrokeTheBuild committed:
Add retry loop for downloading jsch.jar

mwinkel-dev reviewed (approved):
Most recent commit has a retry loop for downloading the Java library that implements SSH in Java programs (aka jsch.jar).   Plus a minor formatting change in the CMakeLists.txt file.

The changes are straightforward, so approving this PR.
